c10/cuda/CUDAException.h

```diff
@@ -14,6 +14,7 @@
   do {                                                     \
     cudaError_t __err = EXPR;                              \
     if (__err != cudaSuccess) {                            \
+      cudaGetLastError();				   \
```

Comment:
@ngimel can you review that this is okay?

Without this, a previously raised error was still lingering and falsely being triggered for a subsequent CUDA call. @colesbury suggested that this is the right thing to do.

Reply:
Yep, it is. It won't work if the error is sticky and context was corrupted, but for other cases if you can recover from RuntimeError thrown by AT_ERROR, you are good to go.